packages/core/tooling/js/vite.ts

```diff
@@ -0,0 +1,132 @@
+import { array, functions, imports, object, exports, type AstTypes, common } from './index.ts';
+
+function exportDefaultConfig(
+	ast: AstTypes.Program,
+	options: {
+		fallback?: AstTypes.Expression | string;
+		ignoreWrapper?: string;
+	} = {}
+): AstTypes.ObjectExpression {
+	const { fallback, ignoreWrapper } = options;
+
+	// Get or create the default export
+	let fallbackExpression: AstTypes.Expression;
+	if (fallback) {
+		fallbackExpression = typeof fallback === 'string' ? common.parseExpression(fallback) : fallback;
+	} else {
+		fallbackExpression = object.create({});
+	}
+
+	const { value: rootObject } = exports.createDefault(ast, { fallback: fallbackExpression });
+
+	// Handle wrapper functions (e.g., defineConfig({})) if ignoreWrapper is specified
+	let configObject: AstTypes.ObjectExpression;
+	if (ignoreWrapper && 'arguments' in rootObject && Array.isArray(rootObject.arguments)) {
+		// Check if this is the wrapper we want to ignore
+		if (
+			rootObject.type === 'CallExpression' &&
+			rootObject.callee.type === 'Identifier' &&
+			rootObject.callee.name === ignoreWrapper
+		) {
```

Comment:
Would suggest inverting those two conditions and bailing out early to improve readability. Maybe even more, but those two should be pretty easy i think.

Apart from that the method looks really good!